test/jest/cookieSyncManager.spec.ts

```diff
@@ -0,0 +1,454 @@
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync('other-mpid', testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {
+                    5: 1234567890
+                },
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        // QUESTION: What is the purpose of this code path?
+        it('should call performCookieSync with mpid if previousMpid and mpid match', () => {
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync(testMPID, testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {},
+                undefined,
+                true,
+                false, 
+            ); 
+        });
+
+        it('should perform a cookie sync if lastSyncDateForModule is past the frequency cap', () => {
+            const now = new Date().getTime();
+
+            // Rev the date by one
+            const cookieSyncDateInPast = now - ( pixelSettings.frequencyCap  + 1 ) * 60 * 60 * 24 * 1000;
+
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                        csd: { 5: cookieSyncDateInPast }
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync(null, testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {
+                    5: cookieSyncDateInPast 
+                },
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        it('should perform a cookie sync if lastSyncDateForModule is past the frequency cap even if csd is empty', () => {
```

Comment:
If CSD is empty, it means it has never been performed yet.  So there shouldn't be a code path for this.  I think this will be part of the code refactor.